spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/IasXsuaaExchangeBroker.java

```diff
@@ -70,27 +88,28 @@ public String getXsuaaToken(Token iasToken) {
 	}
 
 	/**
-	 * Checks value of environment variable 'IAS_XSUAA_XCHANGE_ENABLED'. This value
-	 * determines, whether token exchange between IAS and XSUAA is enabled. If
-	 * IAS_XSUAA_XCHANGE_ENABLED is not provided or with an empty value or with
-	 * value = false, then token exchange is disabled. Any other values are
-	 * interpreted as true.
-	 *
-	 * @return returns true if exchange is enabled and false if disabled
+	 * Resolves the encoded token to Token class
+	 * 
+	 * @param oAuth2Token
+	 *            encoded token
+	 * @return IasToken class
+	 * @throws ParseException
+	 *             if provided Jwt token couldn't be parsed
 	 */
-	public boolean isIasXsuaaXchangeEnabled() {
-		return isIasXsuaaXchangeEnabled;
+	Token decodeToken(String oAuth2Token) throws ParseException {
```

Comment:
maybe not needed? decodeToken and doIasXsuaaXchange comes always together? Furthermore is decodeToken not the right naming ;-) otherwise i would expected a return type DecodedJwt...

formerly we wanted to separate it to decode the token only once: for the isXsuaaToken check and for the token-exchange itself...